sysutils/webdav-backup/src/opnsense/mvc/app/library/OPNsense/Backup/Webdav.php

```diff
@@ -0,0 +1,368 @@
+<?php
+
+/*
+ * Copyright (C) 2018 Deciso B.V.
+ * Copyright (C) 2018 Fabian Franz
+ * Copyright (C) 2023 Nils Stein
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Backup;
+
+use OPNsense\Core\Config;
+use OPNsense\Backup\WebdavSettings;
+
+/**
+ * Class Webdav backup
+ * @package OPNsense\Backup
+ */
+class Webdav extends Base implements IBackupProvider
```

Comment:
you could extend the nextcloud provider to avoid duplication. then you can just add it into the nextcloud plugin

Reply:
Wouldn't it be better to do it the other way around? `Webdav` as base class?

Reply:
You can but then it is more testing effort (need to test nextclould then as well)